### PR TITLE
Fix/login redirect

### LINKED ISSUE #1635: Return to previous page on login
Currently, logging in takes you back to the homepage.

### PR DESCRIPTION
**Pull request by clari182**
- closes #1635 

From any page, if you click Subscribe and then login/signup you should be redirected to the previous page

### CODE REVIEW

File: site/gatsby-site/src/templates/cite.js [hunk 1]
```diff
@@ -230,8 +230,16 @@ function CitePage(props) {
...
+            Please{' '}
+            <Link
+              to={localizePath({
+                path: `/login?redirectTo=${window.location.pathname + window.location.search}`,
+                language: locale,
+              })}
+            >
+              log in
+            </Link>{' '}
+            to subscribe
```
Comment (pdcp1) [inline site/gatsby-site/src/templates/cite.js:242]: Please fix the translations. They are broken after this change with `{' '}`

Spanish version:
image

File: site/gatsby-site/src/templates/entity.js [hunk 2]
```diff
@@ -163,7 +163,11 @@ const EntityPage = ({ pageContext, data, ...props }) => {
       addToast({
         message: (
           <Trans i18n={i18n}>
-            Please <Link to={localizePath({ path: '/login' })}>log in</Link> to subscribe
+            Please{' '}
+            <Link to={localizePath({ path: `/login?redirectTo=${props?.location?.pathname}` })}>
+              log in
+            </Link>{' '}
+            to subscribe
```
Comment (pdcp1) [inline site/gatsby-site/src/templates/entity.js:170]: The same here. Please fix the translations. They are broken after this change with `{' '}`

Spanish version:
image

**Review by pdcp1 — CHANGES_REQUESTED** [review]
Please add tests to this new redirect feature

File: site/gatsby-site/src/templates/entity.js (hunk 2 above)
Comment (pdcp1) [reply site/gatsby-site/src/templates/entity.js:170]: @clari182 Please go to https://deploy-preview-1713--staging-aiid.netlify.app/es/entities/facebook/ and click on `Notificarme de nuevos Incidentes de Facebook` and you will see the toast in English

File: site/gatsby-site/src/templates/cite.js [hunk 3]
```diff
@@ -230,8 +230,16 @@ function CitePage(props) {
...
+            Please{' '}
+            <Link
+              to={localizePath({
+                path: `/login?redirectTo=${props?.location?.pathname}`,
+                language: locale,
+              })}
+            >
+              log in
+            </Link>{' '}
+            to subscribe
```
Comment (pdcp1) [inline site/gatsby-site/src/templates/cite.js:242]: @clari182 Please go to https://deploy-preview-1713--staging-aiid.netlify.app/es/cite/421/ and click on `Notificarme de actualizaciones` and you will see the toast in English

**Review by pdcp1 — CHANGES_REQUESTED** [review]

**Review by pdcp1 — APPROVED** [review]

File: site/gatsby-site/src/templates/cite.js (hunk 3 above)
Comment (pdcp1) [reply site/gatsby-site/src/templates/cite.js:242]: fixed

**Review by cesarvarela — CHANGES_REQUESTED** [review]
I checked twice, but looks like this is still missing tests?

**Comment by clari182** [comment]
@cesarvarela the redirectTo tests were already there in [staging](https://github.com/responsible-ai-collaborative/aiid/blob/400a1febacb7f231fbac4a0d72418f12f1375f7c/site/gatsby-site/cypress/e2e/integration/login.cy.js#L12).
Actually the logic of the redirectTo was already done, what was missing was using it.

**Review by cesarvarela — APPROVED** [review]

**Pull request merged**